scripts/prairie_forest.json

```diff
@@ -0,0 +1,1431 @@
+                        "type": "double"
+                    },
+                    {
+                        "name": "Solanum rostratum",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Sonchus asper",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Sorghastrum nutans",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Unknown species B",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Sporobolus cryptandrus",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Stellaria media",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Symphoricarpos orbiculatus",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Taraxacum officinale",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Teucrium canadense",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Torilis arvensis",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Toxicodendron radicans",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Tragopogon dubius",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Tripsacum dactyloides",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Tridens flavus",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Trifolium pratense",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Trifolium repens",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Ulmus rubra",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Unknown forb A",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Unknown grass A",
+                        "type": "double"
+                    },
+                    {
+                        "name": "Unknown species C",
+                        "type": "double"
+                    },
+                    {
```

Comment:
We can rename the two as
unknown_species_c_first
unknown_species_c_second

Also, remove spaces from all the columns by replacing the spaces with _

Reply:
I replace all the space in the colunms with _ and command `retriever install sqlite prairie-forest `works well. I also want to know the whole steps of testing adding datasets, so I can test by myself when I need to add datasets to retriever next time.
Besides, I want to share you something about the idea "Retriever Out of Memory Functionality". I add code to acquire the size of datasets before downloading, but I am still test the threshold of large datasets. So I cannot pull a request now. I also test more datasets to find the threshold and I find that getting the sizes of all the datasets in retriever is necessary. Now, I am trying to acquire size of datasets in retriever before downloading, especially the size after unzipping. I recommend that adding size of data as part of property of datasets to distinguish large datasets and I will complete this work. Thanks! @henrykironde

Reply:
> whole steps of testing adding datasets, so I can test by myself when I need to add datasets to retriever next time.

we do not have any tools that test the dataset script. We are currently working on one however.
To manually test I usually run retriever on install on all the engines and look at the data if it is consistent in all engines. 

>  I add code to acquire the size of datasets before downloading, but I am still test the threshold of large datasets.

sounds good, let me know when you are ready for me to look at it